### PR TITLE
fix: Convert blurbs to fast for s2s and p2p

### PR DESCRIPTION
**Pull request by isaac-chung**
## Checklist for adding MMTEB dataset

Reason for dataset addition: Address #828 

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py [hunk 1]
```diff
@@ -77,5 +84,28 @@ class BlurbsClusteringP2PFast(AbsTaskClustering):
     )
 
     def dataset_transform(self):
-        ds = clustering_downsample(self.dataset, self.seed)
-        self.dataset = ds
+        ds = dict()
+        for split in self.metadata.eval_splits:
+            labels = list(itertools.chain.from_iterable(self.dataset[split]["labels"]))
```
Comment (imenelydiaker) [inline mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py:89]: `self.dataset[split]["labels"]` is already a list no?

File: mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py [hunk 2]
```diff
@@ -36,7 +43,7 @@ class BlurbsClusteringP2P(AbsTaskClustering):
     )
 
 
-class BlurbsClusteringP2PFast(AbsTaskClustering):
+class BlurbsClusteringP2PFast(AbsTaskClusteringFast):
```
Comment (imenelydiaker) [inline mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py:46]: Need to update the number of samples in line 82?

File: mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py (hunk 1 above)
Comment (isaac-chung) [reply mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py:89]: Ya. My goal is to use code from my previous "convert" PR which works, then refactor all instances in a separate PR.

File: mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py (hunk 2 above)
Comment (isaac-chung) [reply mteb/tasks/Clustering/deu/BlurbsClusteringP2P.py:46]: Done

**Review by imenelydiaker — APPROVED** [review]
Since the refactoring is coming in another PR, let's merge this one 🙂

**Pull request merged**